go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -355,6 +372,70 @@ func (se *Engine) tableWasCreatedOrAltered(ctx context.Context, tableName string
 	return nil
 }
 
+// registerTopics optionally connects the vt_topic metadata on a message table
+// to a map of topic strings. A table can belong to only one topic.
+func (se *Engine) registerTopics() {
+	for _, table := range se.tables {
+		se.registerTopic(table)
+	}
+}
+
+func (se *Engine) registerTopic(ta *Table) {
+	if ta.MessageInfo == nil || ta.MessageInfo.Topic == "" {
+		return
+	}
+
+	topicName := ta.MessageInfo.Topic
+	topicTable, ok := se.tables[topicName]
+	if !ok {
+		// initialize topic table if necessary
+		topicTable = NewTable(topicName)
+		topicTable.TopicInfo = &TopicInfo{
+			Subscribers: make([]*Table, 0, 1),
+		}
+		se.tables[topicName] = topicTable
+		log.Infof("creating topic table '%s'", topicName)
+	} else {
+		// check to see if this table is already registered to the topic
+		// so we don't double register
+		for _, t := range topicTable.TopicInfo.Subscribers {
+			if t.Name == ta.Name {
+				return
+			}
+		}
+	}
+
+	// append this table to the list of subscribed tables to the topic
+	log.Infof("subscribing message table '%s' to topic '%s'", ta.Name.String(), topicName)
+	topicTable.TopicInfo.Subscribers = append(topicTable.TopicInfo.Subscribers, ta)
+}
+
+func (se *Engine) unregisterTopic(ta *Table) {
+	if ta.MessageInfo == nil || ta.MessageInfo.Topic == "" {
+		return
+	}
+
+	topicName := ta.MessageInfo.Topic
+	topicTable, ok := se.tables[topicName]
+	if !ok {
+		panic("topic should have already been created")
+	}
+	// remove this table from the topic
+	for i, t := range topicTable.TopicInfo.Subscribers {
+		// remove the table from the topic
+		if t.Name == ta.Name {
+			log.Infof("unsubscribing message table '%s' from topic '%s'", ta.Name.String(), topicName)
+			topicTable.TopicInfo.Subscribers = append(topicTable.TopicInfo.Subscribers[:i], topicTable.TopicInfo.Subscribers[i+1:]...)
+		}
+	}
+
+	// delete the topic table if there are no more subscribers
```

Comment:
I have a goal to allow for topic inserts even when there are no subscribers, but I'm reserving that for a future PR